### PR TITLE
docs: Update Showcase Dynamic plugins documentation to remove the part related to the old backend system

### PR DESCRIPTION
**Pull request by jesuino**
Fix for [RHIDP-1325](https://issues.redhat.com/browse/RHIDP-1325)

### CODE REVIEW

**Review by schultzp2020 — APPROVED** [review]
/approve
/lgtm

File: showcase-docs/dynamic-plugins.md [hunk 1]
```diff
@@ -44,12 +41,12 @@ These recommended changes to the `package.json` are summarized below:
...
   "dependencies": {
     ...
-    "@backstage/backend-dynamic-feature-service": "0.1.0",
+    "@backstage/backend-dynamic-feature-service": "0.2.3",
     ...
   }
   ...
   "devDependencies": {
-    "@janus-idp/cli": "^1.4.7"
+    "@janus-idp/cli": "^1.7.5"
```
Comment (davidfestal) [inline showcase-docs/dynamic-plugins.md:49]:
```suggestion
    "@janus-idp/cli": "^1.7.7"
```

File: showcase-docs/dynamic-plugins.md [hunk 2]
```diff
@@ -14,14 +14,11 @@ While this package remains in an experimental phase and is a private package in
 
 ### Backend plugins
 
-The backstage showcase application is still using the legacy backend system.
-So to be compatible with the showcase dynamic plugin support, and used as dynamic plugins, existing plugins must be completed code-wise, as well as rebuilt with a dedicated CLI command.
+To be compatible with the showcase dynamic plugin support, and used as dynamic plugins, existing plugins must be completed code-wise, as well as rebuilt with a dedicated CLI command.
```
Comment (davidfestal) [inline showcase-docs/dynamic-plugins.md:17]:
```suggestion
To be compatible with the showcase dynamic plugin support, and used as dynamic plugins, existing plugins must be based on, or compatible with, the new backend system, as well as rebuilt with a dedicated CLI command.
```

File: showcase-docs/dynamic-plugins.md [hunk 3]
```diff
@@ -44,12 +41,12 @@ These recommended changes to the `package.json` are summarized below:
   ...
   "dependencies": {
     ...
-    "@backstage/backend-dynamic-feature-service": "0.1.0",
+    "@backstage/backend-dynamic-feature-service": "0.2.3",
```
Comment (davidfestal) [inline showcase-docs/dynamic-plugins.md:44]: In fact that should not even be necessary in most cases, when using a plugin made for the new backend system.

Comment (davidfestal) [reply showcase-docs/dynamic-plugins.md:44]: So line https://github.com/janus-idp/backstage-showcase/pull/1105/files#diff-8d2e5712d5ef532a983602f4c771597e41d8c082d7801f59f6003fe9f66e2a89L28 can be removed.

Line https://github.com/janus-idp/backstage-showcase/pull/1105/files#diff-8d2e5712d5ef532a983602f4c771597e41d8c082d7801f59f6003fe9f66e2a89R28 should also be removed

Comment (davidfestal) [reply showcase-docs/dynamic-plugins.md:44]: In line 38 you should use:

```
          "export-dynamic": "janus-cli package export-dynamic-plugin ----embed-as-dependencies"
```

File: showcase-docs/dynamic-plugins.md [hunk 4]
```diff
@@ -44,12 +41,12 @@ These recommended changes to the `package.json` are summarized below:
...
     ...
   }
   ...
   "devDependencies": {
-    "@janus-idp/cli": "^1.4.7"
+    "@janus-idp/cli": "^1.7.5"
   },
   ...
   "files": [
```
Comment (davidfestal) [inline showcase-docs/dynamic-plugins.md:52]: This required change can be removed now

File: showcase-docs/dynamic-plugins.md [hunk 5]
```diff
@@ -60,64 +57,11 @@ These recommended changes to the `package.json` are summarized below:
...
-1. The `dynamicPluginInstaller` entry point must be exported in the main `src/index.ts` file:
-
-```ts
-export * from './dynamic/index';
-```
-
-#### Note about the new backend system support
-
-While the Showcase application has not yet integrated the new backend system, the underlying mechanism responsible for discovering and loading dynamic backend plugins is already compatible with both the new and old backend systems.
+#### Creating entry points
```
Comment (davidfestal) [inline showcase-docs/dynamic-plugins.md:60]:
```suggestion
#### Required entry points
```

File: showcase-docs/dynamic-plugins.md [hunk 6]
```diff
@@ -60,64 +57,11 @@ These recommended changes to the `package.json` are summarized below:
...
 
-As the new backend system gains broader acceptance and is implemented in the Janus Showcase, it is advisable to adapt dynamic backend plugins to rely on this new system. Therefore, we **strongly recommend** creating the anticipated entry points for the new backend system (using `createBackendPlugin` or `createBackendModule`) when implementing code changes to enable a backend plugin's dynamic functionality. This proactive step ensures preparedness for the eventual transition to the new backend system.
+It is required to create entry points (using `createBackendPlugin` or `createBackendModule`) to enable a backend plugin's dynamic functionality.
```
Comment (davidfestal) [inline showcase-docs/dynamic-plugins.md:62]:
```suggestion
The new backend system standard entrypoint (created using  `createBackendPlugin()` or `createBackendModule()`) should be exported as the default export of either the main package or of an `alpha` package (if the new backend support is still provided as `apha` APIs).

So this doesn't add any additional requirement on top of the standard plugin development guidelines of the new backend system.
```

**Review by davidfestal — CHANGES_REQUESTED** [review]

**Comment by jesuino** [comment]
Thanks for the review @davidfesta l I made the requested changes.

File: showcase-docs/dynamic-plugins.md [hunk 7]
```diff
@@ -14,110 +14,39 @@ While this package remains in an experimental phase and is a private package in
...
   "scripts": {
     ...
-    "export-dynamic": "janus-cli package export-dynamic-plugin"
-    ...
-  },
-  ...
-  "dependencies": {
-    ...
-    "@backstage/backend-dynamic-feature-service": "0.1.0",
+    "export-dynamic": "janus-cli package export-dynamic-plugin ----embed-as-dependencies"
```
[the remark this answers is not in the document]
Comment (jesuino) [reply showcase-docs/dynamic-plugins.md]: done, thanks!

**Review by davidfestal — APPROVED** [review]

**Pull request merged**